### PR TITLE
Create erb wizard from

### PR DESCRIPTION
**Pull request by iBrooks**
Added an erb form to add a new wizard
removed obsolete api endpoints
removed obsolete api tests
updated show to operate off of id not name

### CODE REVIEW

File: app/controllers/api/v1/wizards_controller.rb [hunk 1]
```diff
@@ -24,6 +24,6 @@ def show
   protected
 
   def wizard_params
-    params.require(:wizard).permit(:name, :description, :img_url, :creator_id)
+    params.require(:wizard).permit(:id, :name, :description, :img_url, :creator_id)
```
Comment (dpickett) [inline app/controllers/api/v1/wizards_controller.rb]: we do not want to allow the user to provide the id / primary key here. this should be left out of strong params

File: app/controllers/wizards_controller.rb [hunk 2]
```diff
@@ -2,4 +2,34 @@ class WizardsController < ApplicationController
   before_action :authenticate_user!, except:[:index, :show]
 
   def index; end
+  def show
+    render :index
+  end
+  def new
```
Comment (dpickett) [inline app/controllers/wizards_controller.rb]: consider a blank line between each method

File: app/controllers/wizards_controller.rb [hunk 3]
```diff
@@ -2,4 +2,34 @@ class WizardsController < ApplicationController
...
+    else
+      redirect_to :index
+    end
+  end
+  def create
+    @wizard = Wizard.new(wizards_params)
+    if current_user
+      if @wizard.save
+        redirect_to wizard_path(@wizard)
+        flash[:notice] = 'Wizards added successfully'
```
Comment (dpickett) [inline app/controllers/wizards_controller.rb]: convention wise, we like the `flash` to come before the redirect.

File: app/javascript/react/components/WizardShow.js [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
+<div>
   <div className="wizard-show-component">
     <h1>{props.name}</h1>
     <img src= {`${props.imgUrl}`} alt=""/>
     <h1>{props.rating}</h1>
+
     <p>{props.description}</p>
   </div>
+
+  <div>
```
Comment (dpickett) [inline app/javascript/react/components/WizardShow.js]: consider indenting this `<div>` for readability

File: app/models/wizard.rb [hunk 5]
```diff
@@ -3,4 +3,7 @@ class Wizard < ApplicationRecord
 
   validates :name, presence: true, uniqueness: true
   validates :description, presence: true
+  def to_param
```
Comment (dpickett) [inline app/models/wizard.rb]: consider a line break above this line

File: db/migrate/20180124144439_add_name_too_users.rb [hunk 6]
```diff
@@ -1,5 +1,5 @@
 class AddNameTooUsers < ActiveRecord::Migration[5.1]
   def change
-    add_column :users, :user_name, :string, null: false
+    add_column :users, :user_name, :string, null: false, default: "user"
```
Comment (dpickett) [inline db/migrate/20180124144439_add_name_too_users.rb:3]: why set a default here? I wouldn't recommend it.

File: spec/features/user_sumbits_new_wizard_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    visit 'wizards/new'
+
+    fill_in 'Name', with: 'Mr McGee'
+    fill_in 'Description', with: "He's a good wizard"
+
+    click_button "Add Wizard"
+
+    expect(page).to have_content 'Wizards added successfully'
+  end
+  scenario 'user submits invalid form' do
```
Comment (dpickett) [inline spec/features/user_sumbits_new_wizard_spec.rb]: these tests should be completed

**Review by dpickett — CHANGES_REQUESTED** [review]

**Comment by prophetsfall** [comment]
We've cleaned this up pull request. Please review.

File: app/controllers/wizards_controller.rb [hunk 8]
```diff
@@ -2,19 +2,22 @@ class WizardsController < ApplicationController
...
       @wizard = Wizard.new
     else
       redirect_to :index
     end
   end
+
   def create
-    @wizard = Wizard.new(wizards_params)
     if current_user
+      @wizard = Wizard.new(wizards_params.merge({ creator_id: current_user.id }))
```
Comment (dpickett) [inline app/controllers/wizards_controller.rb]: I would prefer:

```ruby
@wizard = Wizard.new(wizards_params)
@wizard.user = current_user
```

File: app/views/wizards/new.html.erb [hunk 9]
```diff
@@ -1,8 +1,8 @@
 <h1>New Wizard Form</h1>
 <%= form_for(@wizard) do |f| %>
  <% if @wizard.errors.any? %>
- <div id=“error_explanation”>
-   <h2><%= pluralize(@wizard.errors.count, “error”) %> prohibited this Wizard from being saved:</h2>
+ <div id='error_explanation'>
```
Comment (dpickett) [inline app/views/wizards/new.html.erb]: use double quotes for HTML attribute values

**Review by dpickett — CHANGES_REQUESTED** [review]

**Comment by prophetsfall** [comment]
Requested changes have been completed.

**Comment by prophetsfall** [comment]
Ok, this should be good to go now.

File: app/controllers/wizards_controller.rb [hunk 10]
```diff
@@ -17,12 +17,13 @@ def new
 
   def create
     if current_user
-      @wizard = Wizard.new(wizards_params.merge({ creator_id: current_user.id }))
+      @wizard = Wizard.new(wizards_params)
+      @wizard.creator_id = current_user.id
```
Comment (dpickett) [inline app/controllers/wizards_controller.rb]: you could remove the id here - I find it to be more expressive

`@wizard.creator = current-user`

**Review by dpickett — APPROVED** [review]

**Pull request merged**